src/docs/preferences.md

```diff
@@ -4,191 +4,522 @@ title: Preferences
 
 ```typescript
-const preferences = ctx.container.get(PreferenceService);
+export interface MyExtensionConfiguration {
+    'myExtension.enabled': boolean;
+    'myExtension.timeout': number;
+    'myExtension.logLevel': 'error' | 'warn' | 'info' | 'debug';
+    'myExtension.overridableOption': string;
+}
 ```
 
-In the case of the filesystem, the service is fetched at the beginning for the bindings. There, you can use the onPreferenceChanged method to register a pref changed callback.
+**Note**: This interface is only required if you plan to use preference proxies (step 3). If you only use the `PreferenceService` directly, you can skip this step.
+
+### 3. Create a Preference Proxy (Optional)
+
+A preference proxy provides type-safe access to your preferences. This step is optional - you can also use the `PreferenceService` directly.
 
 ```typescript
+import { createPreferenceProxy, PreferenceProxy, PreferenceService } from '@theia/core/lib/common/preferences';
+
+export const MyExtensionPreferences = Symbol('MyExtensionPreferences');
+export type MyExtensionPreferences = PreferenceProxy<MyExtensionConfiguration>;
 
-constructor(@inject(PreferenceService) protected readonly prefService: PreferenceService
-	prefService.onPreferenceChanged(e => { callback }
+export function createMyExtensionPreferences(preferences: PreferenceService): MyExtensionPreferences {
+    return createPreferenceProxy(preferences, myExtensionPreferenceSchema);
+}
 ```
 
-Where the event received `e` is like this:
+**Alternative**: If you prefer not to use preference proxies, you can work directly with the `PreferenceService` and skip steps 2 and 3.
+
+### 4. Register Your Preferences
+
+Register your preferences schema in your extension's dependency injection module:
 
 ```typescript
-export interface PreferenceChangedEvent {
-    readonly preferenceName: string;
-    readonly newValue?: any;
-    readonly oldValue?: any;
+import { interfaces } from '@theia/core/shared/inversify';
+import { PreferenceContribution, PreferenceService } from '@theia/core/lib/common/preferences';
+
+export const MyExtensionPreferenceContribution = Symbol('MyExtensionPreferenceContribution');
+
+export function bindMyExtensionPreferences(bind: interfaces.Bind): void {
+    // Always required: Register the schema contribution
+    bind(MyExtensionPreferenceContribution).toConstantValue({ schema: myExtensionPreferenceSchema });
+    bind(PreferenceContribution).toService(MyExtensionPreferenceContribution);
+    
+    // Optional: Only if using preference proxy (from step 3)
+    bind(MyExtensionPreferences).toDynamicValue(ctx => {
+        const preferences = ctx.container.get<PreferenceService>(PreferenceService);
+        const contribution = ctx.container.get<PreferenceContribution>(MyExtensionPreferenceContribution);
+        return createMyExtensionPreferences(preferences, contribution.schema);
+    }).inSingletonScope();
```

Comment:
Shouldn't the preferences be integrated in the preference layout somehow? See `preference-layout.ts`?